src/packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -45,6 +45,13 @@ import { mssqlPreparedStatement } from './utils/mssqlPreparedStatement'
 import { tryLoadEnvs } from '@prisma/sdk'
 import { validatePrismaClientOptions } from './utils/validatePrismaClientOptions'
 
+// This is work around see https://github.com/microsoft/TypeScript/issues/17002
```

Comment:
@timsuchanek @Jolg42  Where is the best place to put something like this :-/

Reply:
Hmm instead of patching the global Array constructor types, which could have unintended side-effects, I think we should instead build a little helper function, that uses the `is` keyword to narrow down the type.